### ISSUE TITLE
Performance issue when parsing HOCON

### ISSUE THREAD

**Issue by Lanayx** [akkadotnet/HOCON#276]
My app converts incoming HOCON files to json and I've hit a performance issue with this file:

```
id: "17952a2b-5962-46dd-9183-605cfabe7661-266f04e0"
lang: "en"
session_id: "170ec254-0e34-4c18-85a0-83cd7f61c435"
timestamp: "2020-04-07T10:15:55.921Z"
result {
  source: "agent"
  resolved_query: "kkk"
  action: "input.unknown"
  score: 1.0
  parameters {
  }
  contexts {
    name: "botcop"
    lifespan: 12
    parameters {
      fields {
        key: "tz"
        value {
          string_value: "Europe/Warsaw"
        }
      }
    }
  }
  contexts {
    name: "__system_counters__"
    lifespan: 1
    parameters {
      fields {
        key: "no-input"
        value {
          number_value: 0.0
        }
      }
      fields {
        key: "no-match"
        value {
          number_value: 7.0
        }
      }
    }
  }
  contexts {
    name: "botcopy-timezone"
    lifespan: 11
    parameters {
      fields {
        key: "tz"
        value {
          string_value: "Europe/Warsaw"
        }
      }
    }
  }
  metadata {
    intent_id: "56390419-decd-45e2-b194-5f54f50393fa"
    intent_name: "Default Fallback Intent"
    webhook_used: "false"
    webhook_for_slot_filling_used: "false"
    is_fallback_intent: "true"
  }
  fulfillment {
    speech: "Sorry, I didn't get that. Can you rephrase?"
    messages {
      lang: "en"
      type {
        number_value: 0.0
      }
      speech {
        string_value: "Sorry, I didn't get that. Can you rephrase?"
      }
    }
  }
}
status {
  code: 200
  error_type: "success"
}
```
I'm getting a performance hit, it eats around 500Mb on memory and spends half of minute to convert this single file. Help is appreciated.

What is interesting - if I remove one of _contexts_ sections - there is no performance issue, and everything works normally, but with 3 of them it happens.

I'm using `HoconConfigurationFactory.ParseString` and then calling `config.ToString()`

**Comment by Lanayx** [comment]
One more update - with 4 contexts it's even worse, parsing hangs indefinitetly

**Comment by iron9light** [comment]
Same issue. It tooks 2 min to parse (`HoconParser.Parse`) the hocon below:
```hocon
application: XXX

workflow {
  application: ${application}
  environment: ${env.envName}
  workflowPath: ${workflow.application}/${workflow.environment}/${workflow.workflowType}/${workflow.workflowName}
}

env {
  envName: xxx
}

workflow {
  workflowType: scopeJob
  job {
    delta: "0001-01-01Z" // Default delta == new DateTime(0)
    deltaRevision: 0
    retryTime: 0
    jobPath: "_FORMAT:["${workflow.workflowPath}","${workflow.job.delta}","${workflow.job.deltaRevision}","${workflow.job.retryTime}"]{0}/{1:s}_{2}/{3:0000}"

    vars {
      SubjectAreaResources_SharePath: /xxx/xxx/xxx/
      UETResources_SharePath: /xxx/xxx/xxx/
      Conversion_UserAdActivity_Output_Path: /xxx/xxx/xxx/
      Conversion_Visitization_Output_Path: /xxx/xxx/xxx/
      UET_GoalDSVFilesPath: /xxx/xxx/xxx/
      SA_StripeMapPath: /xxx/xxx/xxx/
      Conversion_Attribution_Output_Path: /xxx/xxx/xxx/
      CDR_C2C_ViewsRelativePath: /xxx/xxx/xxx/
      CDR_C2C_View_Mode: PROD
      C2C_DataMode: PROD
      Conversion_UET_Attribution_FileExpiry: 123
      NumOfHoursPreviousUnAttributedConversion: 123
      NumOfHoursPreviousAdExtensionData: 8
      Conversion_C2CCPGDump_Output_Path: /xxx/xxx/xxx/
    }
  }
}

workflow {
  workflowName: XXXX
  deltaInterval: "01:00:00"
  dependencies: [
    {
      workflowName: ClickUserAdActivity_UET_Conversion
    }
    {
      workflowName: Conversion_UET_Visitization
    }
    {
      workflowName: StripeC2C1HourWorkflow
      offset: "-"${workflow.job.vars.NumOfHoursPreviousAdExtensionData}":0"
    }
  ]
  job {
    jobName: "_FORMAT:["${workflow.job.delta}"]UET_Step2_AppInstall_{0:yyyyMMddHH}"

    parameters: {
      Delta: "_FORMAT:["${workflow.job.delta}"]\"{0:yyyy-MM-dd HH:mm:ss}\""
      ResourcePath: "\""${workflow.job.vars.SubjectAreaResources_SharePath}"\""
      UETResourcesPath: "\""${workflow.job.vars.UETResources_SharePath}"\""
      UserAdActivityOutputPath: "\""${workflow.job.vars.Conversion_UserAdActivity_Output_Path}"\""
      VisitizationOutputPath: "\""${workflow.job.vars.Conversion_Visitization_Output_Path}"\""
      AttributionOutputPath: "\""${workflow.job.vars.Conversion_Attribution_Output_Path}"\""
      C2CViewsPath: "\""${workflow.job.vars.CDR_C2C_ViewsRelativePath}"\""
      C2CViewsMode: "\""${workflow.job.vars.CDR_C2C_View_Mode}"\""
      C2CDataMode: "\""${workflow.job.vars.C2C_DataMode}"\""
      StreamExpiry: "\""${workflow.job.vars.Conversion_UET_Attribution_FileExpiry}"\""
      DSVFilesPath: "\""${workflow.job.vars.UET_GoalDSVFilesPath}"\""
      StripeMapFilesPath: "\""${workflow.job.vars.SA_StripeMapPath}"\""
      NumOfHoursPreviousUnAttributedConversion: "\""${workflow.job.vars.NumOfHoursPreviousUnAttributedConversion}"\""
      NumOfHoursPreviousAdExtensionData: "\""${workflow.job.vars.NumOfHoursPreviousAdExtensionData}"\""
      SA_CPGStripeMapFilePath: "\""${workflow.job.vars.Conversion_C2CCPGDump_Output_Path}"\""
    }
  }
}

"workflow":{"job":{"delta":"2020-04-16T00:00:00+00:00"}}

```

**Comment by iron9light** [comment]
Simple fix for this issue #277 . @Arkatufus @Aaronontheweb could you help to review it?

**Comment by Arkatufus** [comment]
I'm reviewing it right now

**Comment by Lanayx** [comment]
When should we expect a release with this fix?

**Comment by Aaronontheweb** [comment]
We could do one shortly once I've finished reviewing the other PRs.
